server/src/modules/Contest/contest.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, DeleteResult, UpdateResult, InsertResult } from 'typeorm';
+
+import { Contest } from '../../entity/Contest';
+import { IContestDto } from './contest.interface';
+
+@Injectable()
+export class ContestService {
+  constructor(
+    @InjectRepository(Contest)
+    private readonly contestRepository: Repository<Contest>,
+  ) {}
+
+  async findOne(id: number): Promise<Contest> {
+    return await this.contestRepository.findOne(id);
+  }
+
+  async findAll(): Promise<Contest[]> {
+    return await this.contestRepository.find();
+  }
+
+  async create(contest: IContestDto): Promise<InsertResult> {
+    const newContest: Contest = this.contestRepository.create(contest);
+    return await this.contestRepository.insert(newContest);
+  }
+
+  async update(id: number, contest: IContestDto): Promise<UpdateResult> {
+    return await this.contestRepository.update(id, contest);
+  }
+
+  async delete(id: number): Promise<DeleteResult> {
```

Comment:
I believe delete should be soft, as well as in https://github.com/extreme-startup/questionator/pull/58 PR.

Reply:
Soft delete doesn't delete a real entity from the DB, it just changes entity status. So this method should be under the /PUT request type. Contest entity has the isDeleted boolean flag, so users who want to perform soft delete will send:

/PUT //contest/:id
body:  { isDeleted: true }

/DELETE was added for simplifaction of testing, and for possible future needs